tutorials/standalone-tensorflow-raspberry-pi/index.md

```diff
@@ -0,0 +1,168 @@
+
+## Reference Architecture
+
+The following diagram shows the architecture of the solution
+
+![N|Solid](https://storage.googleapis.com/standalone-tensorflow-raspberry-pi/img/standalone_tensorflow_img1.png)
+
+1. Data as pictures, csv files, etc. is originally stored in Google Cloud Storage.
+1. The Tensorflow model training is executed in Cloud ML Engine
+1. The resulting saved model is stored in a bucket in Google Cloud Storage
+1. The saved model is transferred to the IoT Device
+1. Serving of the model (e.g. image recognition) is done locally in the IoT device
+
+## Before you begin
+1. Log into the Raspberry Pi
+
+1. Install required system packages.
+Several packages are required to successfully install gsutil from PyPI. You can install them with the following command:
+    ```
+    sudo apt-get install gcc python-dev python-setuptools libffi-dev
+    ```
+
+1. Install pip
+We recommend using the pip installer. You can install it with the following command:
+    ```
+	sudo apt-get install python-pip
+	```
+
+1. Install gsutil from PyPI
+To install gsutil from PyPI, use the following command:
+	```
+    sudo pip install gsutil
+    ```
+
+## Installing Tensorflow and its dependencies
+1. Install Tensorflow for Raspberry Pi
+	```
+    wget https://github.com/lhelontra/tensorflow-on-arm/releases/download/v1.8.0/tensorflow-1.8.0-cp27-none-linux_armv7l.whl
+	sudo pip install tensorflow-1.8.0-cp27-none-linux_armv7l.whl
+	sudo pip uninstall mock
+	sudo pip install mock
+	```
+
+1. Install fswebcam and fbi
+    ```
+	sudo apt-get install fswebcam
+	sudo apt-get install fbi
+	```
+
+## Downloading a trained saved model and the script for serving
+1. Create a directory where to deploy the model, e.g. “/flowers_model”
+	```
+    mkdir /tf_server/flowers_model/1
+    ```
+
+1. Download the Model
+	```
+    gsutil cp gs://standalone-tensorflow-raspberry-pi/* /tf_server/flowers_model/1/
+    ```
+
+
+1. Open the Python file that does the serving and ustomize the model directory if needed
+    ```
+	nano /standalone-tensorflow-raspberry-pi/tf_server.py
+	```
+
+
+	The variable “model_dir” in line 16 of the python file “tf_server.py” identifies the directory where the model is saved. You can edit this if you download a model in a different location.
+
+    ``` 
+    print '--- importing packages'
+
+	from tensorflow.contrib import predictor
+	import base64
+	import sys
+	import json
+	import subprocess
+	import datetime
+
+	“”” 
+	change in the line below the directory where you hosted your model
+	this is the directory where the .pb file and ‘variables’ directory
+	are hosted. Therefore make sure to add the ‘/1’ for the version of 
+	the model if necessary
+	“””
+	model_dir = '/tf_server/flowers_model/1'
+    ```
+
+
+## Running the Tensorflow Server
+Type the line below to start the Tensorflow server. Note that loading the Tensorflow libraries and Tensorflow saved model may take several minutes to load.
```

Comment:
There's some redundancy in the second sentence in this paragraph.

I'd write this as "Note that loading the TensorFlow libraries and saved model may take several minutes."

Reply:
Done.